repose-aggregator/components/filters/header-id-mapping/src/main/scala/org/openrepose/filters/headeridmapping/HeaderIdMappingFilter.scala

```diff
@@ -0,0 +1,97 @@
+ */
+
+package org.openrepose.filters.headeridmapping
+
+import java.net.URL
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.HttpServletRequest
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.http.PowerApiHeader
+import org.openrepose.commons.utils.servlet.http.HttpServletRequestWrapper
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.headeridmapping.config.{HttpHeader, HeaderIdMappingConfig}
+
+import scala.collection.JavaConverters._
+
+@Named
+class HeaderIdMappingFilter @Inject()(configurationService: ConfigurationService)
+  extends Filter with UpdateListener[HeaderIdMappingConfig] with LazyLogging {
+
+  import HeaderIdMappingFilter._
+
+  private var configurationFile: String = DEFAULT_CONFIG
+  private var initialized = false
+  private var configuredHeaders: Iterable[HttpHeader] = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    logger.trace("Header Identity Mapping filter initializing...")
+    configurationFile = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+
+    logger.info(s"Initializing Header Identity Mapping filter using config $configurationFile")
+    val xsdUrl: URL = getClass.getResource(SCHEMA_FILE_NAME)
+    configurationService.subscribeTo(filterConfig.getFilterName, configurationFile, xsdUrl, this, classOf[HeaderIdMappingConfig])
+
+    logger.trace("Header Identity Mapping filter initialized.")
+  }
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    val wrappedRequest = new HttpServletRequestWrapper(servletRequest.asInstanceOf[HttpServletRequest])
+
+    // find the first configured "user-header" that exists in the request and use that config section to populate the user/group headers
+    configuredHeaders.toStream.flatMap(extractUserGroup(wrappedRequest, _)).headOption.foreach { userGroupValues =>
+      wrappedRequest.addHeader(PowerApiHeader.USER.toString, userGroupValues.user, userGroupValues.quality)
+      userGroupValues.group.foreach(wrappedRequest.addHeader(PowerApiHeader.GROUPS.toString, _, userGroupValues.quality))
+    }
+
+    filterChain.doFilter(wrappedRequest, servletResponse)
+  }
+
+  private def extractUserGroup(request: HttpServletRequestWrapper, configuredHeader: HttpHeader): Option[UserGroupValues] =
+    getFirstHeaderValue(request.getHeader(configuredHeader.getUserHeader))
+      .map(UserGroupValues(_, getFirstHeaderValue(request.getHeader(configuredHeader.getGroupHeader)), configuredHeader.getQuality))
+
+  // carry-over behavior from the old filter - be sure we only get the first value of the header
+  def getFirstHeaderValue(header: String): Option[String] = Option(header).map(_.split(",", 2)(0).trim).filterNot(_.isEmpty)
```

Comment:
This is unneccessary. The wrapper already will do splitting, and it maintains ordering. Call `getSplittableHeaders` or `getSplittableHeaderScala`.